packages/application-generic/src/services/auth/auth.service.ts

```diff
@@ -12,48 +19,65 @@ import {
   EnvironmentEntity,
   IApiKey,
 } from '@novu/dal';
-import { AuthProviderEnum, IJwtPayload, ISubscriberJwt, MemberRoleEnum, SignUpOriginEnum } from '@novu/shared';
 import {
-  AnalyticsService,
-  Instrument,
-  PinoLogger,
-  CachedEntity,
+  AuthProviderEnum,
+  IJwtPayload,
+  ISubscriberJwt,
+  MemberRoleEnum,
+  SignUpOriginEnum,
+} from '@novu/shared';
+
+import { PinoLogger } from '../../logging';
+import { AnalyticsService } from '../analytics.service';
+import { ApiException } from '../../utils/exceptions';
+import { Instrument } from '../../instrumentation';
+import {
+  CreateUser,
+  SwitchEnvironment,
+  SwitchEnvironmentCommand,
+  SwitchOrganization,
+  SwitchOrganizationCommand,
+  CreateUserCommand,
+} from '../../usecases';
+import {
+  buildAuthServiceKey,
   buildEnvironmentByApiKey,
   buildSubscriberKey,
   buildUserKey,
-  buildAuthServiceKey,
-} from '@novu/application-generic';
-
-import { CreateUserCommand } from '../../user/usecases/create-user/create-user.dto';
-import { CreateUser } from '../../user/usecases/create-user/create-user.usecase';
-import { SwitchEnvironmentCommand } from '../usecases/switch-environment/switch-environment.command';
-import { SwitchEnvironment } from '../usecases/switch-environment/switch-environment.usecase';
-import { SwitchOrganization } from '../usecases/switch-organization/switch-organization.usecase';
-import { SwitchOrganizationCommand } from '../usecases/switch-organization/switch-organization.command';
-import { normalizeEmail } from '../../shared/helpers/email-normalization.service';
-import { ApiException } from '../../shared/exceptions/api.exception';
+  CachedEntity,
+} from '../cache';
+import { normalizeEmail } from '../../utils/email-normalization';
 
 @Injectable()
 export class AuthService {
   constructor(
     private logger: PinoLogger,
     private userRepository: UserRepository,
     private subscriberRepository: SubscriberRepository,
+    @Inject(forwardRef(() => CreateUser))
```

Comment:
If moving this to App Generic creates a circular dependency we are doing something wrong. I can help you to find out the guilty relationship.

Reply:
Actually, I am not sure why this one happened it did not have circular dependency until I merged from next yesterday. which is very weird.